src/main/kotlin/org/rust/cargo/project/model/impl/CargoSyncTask.kt

```diff
@@ -421,4 +427,25 @@ private fun <T, R> BuildProgress<BuildProgressDescriptor>.runWithChildProgress(
     }
 }
 
+/**
+ * This cache is really used *only* in unit tests.
+ *
+ * We assume that `sysroot`, a `list of targets`, `--print cfg` and `stdlib`
+ * are not changed until the version of rustc is changed, so we can cache these values for all tests.
+ * The cache significantly speeds up heavy tests with a full toolchain (`RsWithToolchainTestBase`)
+ */
+private object UnitTestRustcCache {
+    private val cache: ConcurrentHashMap<Pair<RustcVersion, Class<*>>, Optional<Any>> = ConcurrentHashMap()
+
+    inline fun <reified T> cached(rustcVersion: RustcVersion?, cacheIf: () -> Boolean = { true }, noinline computation: () -> T): T {
+        if (!isUnitTestMode || rustcVersion == null || !cacheIf()) return computation()
+        return cachedInner(rustcVersion, T::class.java, computation)
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    fun <T> cachedInner(rustcVersion: RustcVersion, cls: Class<T>, computation: () -> T): T {
```

Comment:
Private

Reply:
It can't be private because `cached()` is `inline`, haha

Reply:
It can because `UnitTestRustcCache` is private.
But If we make `UnitTestRustcCache` public, we need to move all `isUnitTestMode` check into it